src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs

```diff
@@ -781,6 +781,7 @@ public int EncodeToUtf8(Span<byte> destination)
 
         public override int GetHashCode() => Value;
 
+#if !MICROSOFT_BCL_MEMORY
```

Comment:
Why are these `#if !MICROSOFT_BCL_MEMORY` rather than `#if NET` or `#if SYSTEM_PRIVATE_CORELIB`? I'd prefer it if the corelib files didn't need to be aware of Microsoft.Bcl.Memory by name.